### DISCUSSION TITLE
disabled tls occurring at startup regardless of settings

### DISCUSSION THREAD

**Discussion by yarikoptic** [photoprism/photoprism#4693] [General]
My case is very similar to the
- https://github.com/photoprism/photoprism/discussions/4325 by @arbourp

so I stole the title, but 

- I fail to make it work (even with full restart of containers)
- it is a bit frustrating that in both cases it seems that logging is missing to describe the behavior (no TLS despite settings) - may be smth with certificates? permissions?

In my case I generated certificates using router's integration with let's encrypt, made them available within container (using podman) for photoprism, followed https://docs.photoprism.app/getting-started/using-https/#3-custom-certificate configuration, forwarded a different WAN port to my laptop's default to prism 2342 (mapped as is into container)

```shell
❯ podman exec -it photoprism bash -c export | grep -E 'PHOTOPRISM.*(TLS|URL|INIT|HTTP_PORT)' | sed -E 's,//(.*)(\.asuscomm.com:)([0-9]+)/,//SENSORED\2PORT,g'
declare -x PHOTOPRISM_DEFAULT_TLS="true"
declare -x PHOTOPRISM_DISABLE_TLS="false"
declare -x PHOTOPRISM_HTTP_PORT="2342"
declare -x PHOTOPRISM_INIT="https"
declare -x PHOTOPRISM_SITE_URL="https://SENSORED.asuscomm.com:PORT"
declare -x PHOTOPRISM_TLS_CERT="site.crt"
declare -x PHOTOPRISM_TLS_KEY="site.key"
```

```shell
❯ podman exec -it photoprism ls -l /photoprism/storage/config/certificates/
total 16
-rw-rw-r-- 1 photoprism photoprism 1598 Dec 23 18:25 cacert.pem
-rw------- 1 photoprism photoprism 1675 Dec 23 18:25 cakey.pem
-rw-r--r-- 1 photoprism photoprism 2869 Dec 26 18:37 cert.pem
-rw------- 1 photoprism photoprism  227 Dec 26 18:37 key.pem
```

```shell
❯ podman logs photoprism 2>&1 | grep -i tls
Creating a default HTTPS/TLS certificate.
Generating self-signed tls certificate...
time="2024-12-31T17:12:46Z" level=info msg="server: disabled auto tls"
```

and nothing more (and I have "trace" level enabled), but photoprism still just talks `http` not `https`  (the same on "secret" external url)

```shell
❯ curl http://localhost:2342
<a href="/library/login">Temporary Redirect</a>.

❯ curl https://localhost:2342
curl: (35) GnuTLS, handshake failed: An unexpected TLS packet was received.
```

edit: I now have also done a complete clean (`git clean`'ed everything - DB etc) "re-start/import" -- the same story, so it is clearly not about some stale state somewhere.

**Comment by lastzero** [comment]
The file names don't match!? 🤔

**Comment by yarikoptic** [comment]
uff, I should put laptop away.  I spotted finally that config has 

```
declare -x PHOTOPRISM_TLS_CERT="site.crt"
declare -x PHOTOPRISM_TLS_KEY="site.key"
```

whenever from router I got

```
-rw-r--r-- 1 photoprism photoprism 2869 Dec 26 18:37 cert.pem
-rw------- 1 photoprism photoprism  227 Dec 26 18:37 key.pem
```

I have converted (if it is a right way to do it):

```shell
❯ openssl ec -outform der -in key.pem -out site.key
read EC key
writing EC key
❯ openssl x509 -outform der -in cert.pem -out site.crt
❯ ls -lt site*
-rw-rw-r-- 1 yoh yoh 920 Dec 31 12:54 site.crt
-rw------- 1 yoh yoh 121 Dec 31 12:53 site.key
```

but seems to be the same result... but issue somewhat remains that logs do not hint on what is misconfigured etc. I would suspect that TLS configs pointing to those files are simply ignored since there is "Creating a default HTTPS/TLS certificate." which shouldn't be needed ... but then again -- why not TLS altogether?

**Comment by yarikoptic** [comment]
having learned how to read those filenames, and seeing that converted files are not actually "digested well" (logs included "failed to find any PEM data in certificate input") I just pointed to those .pem files and it worked:

```
❯ grep PHOTOPRISM_TLS_ docker-compose.yml
      PHOTOPRISM_TLS_CERT: "cert.pem"
      PHOTOPRISM_TLS_KEY: "key.pem"
```

sorry for the noise.
